hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -285,15 +283,64 @@ client tracer point queue headState callback nodeToClientV =
     , cStateQueryCodec
     } = defaultCodecs nodeToClientV
 
-chainSyncClient ::
+data ChainSyncHandler m = ChainSyncHandler
+  { onRollForward :: Block -> m ()
+  , onRollBackward :: Point Block -> m ()
+  }
+
+newChainSyncHandler ::
   forall m.
-  (MonadSTM m, MonadThrow m) =>
+  (MonadSTM m) =>
+  -- | Tracer for logging
   Tracer m DirectChainLog ->
-  ChainCallback Tx m ->
+  -- | Chain callback
+  (ChainEvent Tx -> m ()) ->
+  -- | On-chain head-state.
   TVar m SomeOnChainHeadState ->
+  -- | A chain-sync handler to use in a local-chain-sync client.
+  m (ChainSyncHandler m)
+newChainSyncHandler tracer callback headState = do
+  pure $
```

Comment:
Why is this function monadic? AFAIS this could be a pure (smart) constructor of `ChainSyncHandler`

Reply:
No :no_mouth: , but the monadic nature was indeed done in anticipation.

Reply:
OKAY! It's not needed in the end.